shell/common/platform_view.h

```diff
@@ -796,6 +796,22 @@ class PlatformView {
       std::unique_ptr<AssetResolver> updated_asset_resolver,
       AssetResolver::AssetResolverType type);
 
+  //--------------------------------------------------------------------------
+  /// @brief      Creates a Surface suitable for raster snapshotting. The
+  ///             rasterizer will request this surface if no on screen surface
+  ///             is currently available and a snapshot has been requested
+  ///             by the framework, e.g. if `Scene.toImage` or `Picture.toImage`
+  ///             are called while the application is in the background.
+  ///
+  ///             Not all backends support this kind of surface usage, and the
+  ///             default implementation returns nullptr. Platforms should
+  ///             override this if they can support GPU operations in the
+  ///             background and support GPU resource context usage.
+  ///
+  ///             This is the only public method of this interface called on the
+  ///             raster task runner.
+  virtual std::unique_ptr<Surface> CreateRasterSnapshotSurface();
```

Comment:
Is there a way to avoid having a method on `PlatformView` which is a special case that is not called on the platform thread?

One potential option would be creating another interface similar to `ExternalViewEmbedder` that is instantiated by the `PlatformView` and then given to the `Rasterizer` by the `Shell`.

Like `AndroidExternalViewEmbedder`, this object could hold a reference to the `AndroidContext` and use that to create the pbuffer surface.

Reply:
I like this idea. I'll try to do this - I think it'll get rid of some of the funky threading concerns around accessing the platform view.

Reply:
Ok, latest commit has these changes. This gets rid of the threading concern/note that I had in platform view before. @chinmaygarde FYI